examples/ndn-ping/ndn_ping.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (C) 2016 Wentao Shang
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       NDN ping client and server implemetation
+ *
+ * @author      Wentao Shang <wentaoshang@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <inttypes.h>
+
+#include "thread.h"
+#include "random.h"
+
+#include "ndn-riot/app.h"
+#include "ndn-riot/ndn.h"
+#include "ndn-riot/encoding/name.h"
+#include "ndn-riot/encoding/interest.h"
+#include "ndn-riot/encoding/data.h"
+#include "ndn-riot/msg-type.h"
+
+static ndn_app_t* handle = NULL;
+
+/* static const unsigned char key[] = { 'd', 'u', 'm', 'm', 'y', 'k', 'e', 'y' }; */
+
+static const uint8_t ecc_key_pri[] = {
```

Comment:
Are those keys always the same or should they be set by the user ? If it's the second case, then you should find a way to make them configurable (at build time or by using a setter function)

Reply:
Well, the user of an android don't choose the keys of his apps.
And if you want build-time configuration, I would do it the current way.